neuraxle/steps/flow.py

```diff
@@ -150,44 +149,49 @@ def __init__(self, wrapped: BaseStep):
         TrainOrTestOnlyWrapper.__init__(self, wrapped=wrapped, is_train_only=False)
 
 
-class IfExecutionPhaseIsThenDo(ForceHandleOnlyMixin, MetaStep): # TODO : CHange this for a more general If Then DO
+
+class ExecuteIf(HandleOnlyMixin, MetaStep):
+    def __init__(self, condition_function: Callable, wrapped: BaseStep):
+        MetaStep.__init__(self, wrapped)
+        HandleOnlyMixin.__init__(self)
+        self.condition_function = condition_function
+
+    def _fit_data_container(self, data_container: DataContainer, context: ExecutionContext):
+        if self.condition_function(self, data_container, context):
+            return MetaStep._fit_data_container(self, data_container, context)
+        return self
+
+    def _fit_transform_data_container(self, data_container: DataContainer, context: ExecutionContext):
+        if self.condition_function(self, data_container, context):
+            return MetaStep._fit_transform_data_container(self, data_container, context)
+        return self, data_container
+
+    def _transform_data_container(self, data_container: DataContainer, context: ExecutionContext):
+        if self.condition_function(self, data_container, context):
+            return MetaStep._transform_data_container(self, data_container, context)
+        return data_container
+
+class IfExecutionPhaseIsThen(ExecuteIf):
     """
     If, at runtime, the execution phase is the same as the one given to the constructor, then execute wrapped step.
 
     By default, will raise an error if the execution phase is not specified in the context.
     Steps which implement ForceHandleMixin create context with unspecified phase on fit, fit_transform and transform call.
     """
 
+
     def __init__(self, phase: ExecutionPhase, wrapped: BaseTransformer, raise_if_phase_unspecified: bool = True):
```

Comment:
We could add such assertion a the root of the pipeline, but it doesn't remove the necessity of a local assertion too. I created this specifically because we have a the ForceHandleMixin which creates context (with unspecified phase) and thus there is a potential for (wrongly) changing phase within the execution of a pipeline